contracts/SecurityTokenRegistry.sol

```diff
@@ -741,12 +739,11 @@ contract SecurityTokenRegistry is ISecurityTokenRegistry, EternalStorage {
     }
 
     /**
-     * @notice Changes the PolyToken address. Only Polymath.
-     * @param _newAddress is the address of the polytoken.
+     * @notice Stores the contract addresses of other key contracts from the PolymathRegistry
      */
-    function updatePolyTokenAddress(address _newAddress) external onlyOwner {
-        require(_newAddress != address(0), "Invalid address");
-        set(POLYTOKEN, _newAddress);
+    function updateFromRegistry() external onlyOwner {
+        address _polymathRegistry = getAddress(Encoder.getKey("polymathRegistry"));
+        set(Encoder.getKey("polyToken"), IPolymathRegistry(_polymathRegistry).getAddress("PolyToken"));
```

Comment:
this is an extra step need to perform after calling `initialize` function. As we will never call initialize function again to set the variables because it already presents in the proxy storage. To make thing consistent you can create an internal function of `updateFromRegistry()` that can be called by the `initialize()` function so we can get rid of the extra step of setting up the PolyToken address (If we ever deploy the whole proxy contract again).  
One more thing use constant `POLYTOKEN` variable instead of `Encoder.getKey("polyToken")`